Components/Browse.php

```diff
@@ -2,55 +2,60 @@
 
 namespace EzSystems\PlatformUIBundle\Components;
 
+use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\Routing\RouterInterface;
 
-class Browse implements Component, \JsonSerializable
+class Browse extends Component
 {
-    protected $locationId = false;
+    protected $router;
 
-    protected $locationRestId = false;
+    protected $request;
 
     // passing the router is needed because at the moment the UDW only supports
     // a REST Location id as the starting Location id, so this id is build there
-    public function __construct($request, RouterInterface $router)
+    public function __construct(RequestStack $stack, RouterInterface $router)
     {
-        if ($request->attributes->get('location')) {
-            $this->locationId = $request->attributes->get('location')->id;
-            $this->locationRestId = $router->generate(
-                'ezpublish_rest_loadLocation',
-                ['locationPath' => trim($request->attributes->get('location')->pathString, '/')]
-            );
-        }
+        $this->request = $stack->getMasterRequest();
```

Comment:
I'm not sure you should do that. The point of the RequestStack is to give you the request when you need it. Even though it probably won't matter much, you may wanna call `getMasterRequest()` at some other point.